src/switch_rtp.c

```diff
@@ -2199,6 +2272,9 @@ static int check_rtcp_and_ice(switch_rtp_t *rtp_session)
 				rtcp_generate_report_block(rtp_session, rtcp_report_block, nack_dup);
 				rtp_session->rtcp_send_msg.header.count = 1; /* reception report block count */
 				stats->sent_pkt_count = 0;
+				if (switch_channel_var_true(switch_core_session_get_channel(rtp_session->session), SWITCH_RTCP_FIRE_EVENTS)) {
```

Comment:
you'll want to check a flag here, not a channel var. Also, this should be done only for AUDIO .

Reply:
> you'll want to check a flag here, not a channel var. Also, this should be done only for AUDIO .

Why only audio?  The rtcp sender report is sent either way, so should the esl event too?
Regarding gating the event on a flag, what is the intention?  If we want this gated by flag, what sets or clears that flag?

Your comment from Dec 2th:
> call of sendRTCPEvent() should be under cond 'if (switch_channel_var_true(session->channel, "fire_rtcp_events")'

By comparison, firing of SWITCH_EVENT_RECV_RTCP_MESSAGE in switch_core_media.c happens in any case.

Reply:
you're calling switch_core_session_get_channel() from where you shouldn't call it. it's got an assert in it, it will fail the assert and kill FS when there's no channel.  Also, switch_channel_var_true() calls switch_channel_get_variable_dup() which mutex locks and it will lock for every RTCP SR. That's why I suggested using a RTP session flag, you can see the pattern across the whole check_rtcp_and_ice() function.  For video the RTCP SR is much more often than for audio. If you need for video too add them on different flags at least (enable them separately). You'll get probably hundreds or even thousands of these events on a typical 1 hour video call.  To what purpose ?

Reply:
@dragos-oancea the point of receiving all of the RTCP events on the bus is to be able to send them to a system (e.g. Homer) to be able to monitor quality of all rtp session of all calls/channels.  This occurs currently for RTCP_Receive, we are just duplicating for SEND.  The other reason is that for many calls, there will be encryption, and therefore the packet capture method is not possible.  Also for many client (e.g. web browsers), it may not be possible to capture the RTCP events client side for monitoring.

Reply:
just RTCP SR will not help very much with monitoring video quality, you'll have to do the same for PLI/FIR/NACK.     Just RTCP SR will give you an idea about packet loss and RTT, and the price to pay on generating and processing these events is too high, imo.

Reply:
@dragos-oancea now triggering event only for audio, or for video if rtp session flag set (which gets set if channel var is set)